cudax/test/cuco/utility/hash_test.cu

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <thrust/device_vector.h>
+#include <thrust/execution_policy.h>
+#include <thrust/host_vector.h>
+#include <thrust/logical.h>
+
+#include <cuda/std/array>
+#include <cuda/std/cstddef>
+#include <cuda/std/functional>
+#include <cuda/std/limits>
+#include <cuda/std/span>
+
+#include <cuda/experimental/__cuco/hash_functions.cuh>
+
+#include <testing.cuh>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+template <int32_t Words>
+struct large_key
+{
+  constexpr _CCCL_HOST_DEVICE large_key(int32_t value) noexcept
+  {
+    for (int32_t i = 0; i < Words; ++i)
+    {
+      data_[i] = value;
+    }
+  }
+
+private:
+  int32_t data_[Words];
+};
+
+template <typename Hash, typename Key, typename... HashConstructorArgs>
+static __host__ __device__ bool
+check_hash_result(Key const& key, std::uint32_t expected, HashConstructorArgs&&... hash_constructor_args) noexcept
+{
+  Hash h(::cuda::std::forward<HashConstructorArgs>(hash_constructor_args)...);
+
+  cuda::std::array<Key, 1> arr_keys = {key};
+
+  return (h(key) == expected)
+      && (h(cuda::std::span<Key>(thrust::raw_pointer_cast(arr_keys.data()), arr_keys.size())) == expected);
+}
+
+template <typename TestFn, typename... Args>
+__global__ void test_fn_on_device(TestFn test_fn, Args... args)
+{
+  test_fn(args...);
+}
+
+struct test_xxhash_32
+{
+  template <typename OutputIter>
+  _CCCL_HOST_DEVICE void operator()(OutputIter result)
+  {
+    int i = 0;
+
+    result[i++] = check_hash_result<cudax::cuco::Hash<char>>(static_cast<char>(0), 3479547966, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<char>>(static_cast<char>(42), 3774771295, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<char>>(static_cast<char>(0), 2099223482, 42);
+
+    result[i++] = check_hash_result<cudax::cuco::Hash<int32_t>>(static_cast<int32_t>(0), 148298089, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int32_t>>(static_cast<int32_t>(0), 2132181312, 42);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int32_t>>(static_cast<int32_t>(42), 1161967057, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int32_t>>(static_cast<int32_t>(123456789), 2987034094, 0);
+
+    result[i++] = check_hash_result<cudax::cuco::Hash<int64_t>>(static_cast<int64_t>(0), 3736311059, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int64_t>>(static_cast<int64_t>(0), 1076387279, 42);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int64_t>>(static_cast<int64_t>(42), 2332451213, 0);
+    result[i++] = check_hash_result<cudax::cuco::Hash<int64_t>>(static_cast<int64_t>(123456789), 1561711919, 0);
+
+#if _CCCL_HAS_INT128()
+    result[i++] = check_hash_result<cudax::cuco::Hash<__int128_t>>(static_cast<__int128_t>(123456789), 1846633701, 0);
```

Comment:
This will probably break the tests for any platform that does not provide int128

Also we should probably also tests some unsigned types